subprojects/gradle/cd/src/main/kotlin/com/avito/ci/CiStepsPlugin.kt

```diff
@@ -0,0 +1,65 @@
+package com.avito.ci
+
+import com.avito.utils.gradle.BuildEnvironment
+import com.avito.utils.gradle.buildEnvironment
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.kotlin.dsl.container
+import org.gradle.kotlin.dsl.register
+
+class CiStepsPlugin : Plugin<Project> {
+
+    private val taskGroup = "ci"
+
+    override fun apply(project: Project) {
+
+        val buildContainer = project.container { name ->
+            BuildStepListExtension(name, project.objects)
+        }
+
+        project.extensions.add("builds", buildContainer)
+
+        buildContainer.registerPredefinedBuildTasks()
```

Comment:
Why do we need to create tasks in advance when plugin already creates them dynamically?

Reply:
for backward compatibility; will add it in Avito after release, and remove here on next one

Reply:
oh i see your point; will test on Avito project and update